### DISCUSSION TITLE
when pause button is clicked the layer changes/ skips by 10+

### DISCUSSION THREAD

**Discussion by goliathdj** [mainsail-crew/mainsail#1984] [Help]
im currently running Klipper on raspberry PI 4, paired with the ender S1 pro.
the issue that i have started experiencing is that when i press the pause button the layer count shifts by an increment of 10 to 15 layers. example. when pause is pressed it will change from "Layer 10 of 100" to "Layer 21 of 100".
this causes the print to now continue on air. i even did a reinstall of the image and the same thing still happens. is there something that i am missing ?

**Comment by meteyou** [comment]
Thx for opening a discussions post for your issue.

When you see changing layer numbers, when you press/execute PAUSE, then you only have the calculated layer numbers from Mainsail and not real layer numbers. This is just a fallback in Mainsail, when you don't use the layer numbers macros in your slicer.

To have "real layer numbers", you can follow this guide to add the support in your slicer: https://docs.mainsail.xyz/overview/slicer/prusaslicer#layer-count

But this issue don't explain, why it will continue on a higher layer number. I think you use a wrong PAUSE/RESUME macro, which create a issue like that. Pls upload a klippy.log to check your macros.

**Reply by goliathdj** [reply]
![image_2024-08-26_211054330](https://github.com/user-attachments/assets/b9168f31-ce4a-4e66-9b9a-02e6e1d19987)
![Screenshot 2024-08-26 211024](https://github.com/user-attachments/assets/29ef0cf7-7305-45ef-9655-6bc778ce0173)
[klippy.log](https://github.com/user-attachments/files/16752736/klippy.log)
[moonraker.log](https://github.com/user-attachments/files/16752737/moonraker.log)


attached is the screenshots and logs , i'm dumbfounded not even sure i know how to explain it as this is a clean install of the OS

**Reply by meteyou** [reply]
if it just show the wrong layer number, during the pause, it is just your gcode, because it doesn't contain layer numbers. if your gcode doesn't have layer numbers, Mainsail just calculate it and then 2,3mm above from your bed is just the layer number 12 with your layerheight.

**Reply by goliathdj** [reply]
so chances are this could be a setting on cura ? 
any idea where to find this setting? 
@meteyou

**Reply by meteyou** [reply]
yep, just check the mainsail docs: https://docs.mainsail.xyz/overview/slicer/cura#best-practise-g-codes

i would recommend you, to use the script from pedro. this includes all important gcodes for you.
